crates/ruff/src/rules/pycodestyle/rules/literal_comparisons.rs

```diff
@@ -154,54 +154,64 @@ pub(crate) fn literal_comparisons(
-                if checker.patch(diagnostic.kind.rule()) {
-                    bad_ops.insert(0, Cmpop::Is);
-                }
-                diagnostics.push(diagnostic);
-            }
-            if matches!(op, Cmpop::NotEq) {
-                let diagnostic = Diagnostic::new(NoneComparison(op.into()), comparator.range());
-                if checker.patch(diagnostic.kind.rule()) {
-                    bad_ops.insert(0, Cmpop::IsNot);
-                }
-                diagnostics.push(diagnostic);
-            }
-        }
-
-        if check_true_false_comparisons {
-            if let Expr::Constant(ast::ExprConstant {
-                value: Constant::Bool(value),
-                kind: None,
-                range: _,
-            }) = comparator
+        if let Some(op) = EqCmpop::try_from(op) {
+            if check_none_comparisons
+                && matches!(
+                    comparator,
+                    Expr::Constant(ast::ExprConstant {
+                        value: Constant::None,
+                        kind: None,
+                        range: _
+                    })
+                )
             {
-                if matches!(op, Cmpop::Eq) {
-                    let diagnostic =
-                        Diagnostic::new(TrueFalseComparison(*value, op.into()), comparator.range());
-                    if checker.patch(diagnostic.kind.rule()) {
-                        bad_ops.insert(0, Cmpop::Is);
+                match op {
+                    EqCmpop::Eq => {
+                        let diagnostic = Diagnostic::new(NoneComparison(op), comparator.range());
+                        if checker.patch(diagnostic.kind.rule()) {
+                            bad_ops.insert(0, Cmpop::Is);
+                        }
+                        diagnostics.push(diagnostic);
+                    }
+                    EqCmpop::NotEq => {
+                        let diagnostic = Diagnostic::new(NoneComparison(op), comparator.range());
+                        if checker.patch(diagnostic.kind.rule()) {
+                            bad_ops.insert(0, Cmpop::IsNot);
+                        }
+                        diagnostics.push(diagnostic);
                     }
-                    diagnostics.push(diagnostic);
                 }
-                if matches!(op, Cmpop::NotEq) {
-                    let diagnostic =
-                        Diagnostic::new(TrueFalseComparison(*value, op.into()), comparator.range());
-                    if checker.patch(diagnostic.kind.rule()) {
-                        bad_ops.insert(0, Cmpop::IsNot);
+            }
+
+            if check_true_false_comparisons {
+                if let Expr::Constant(ast::ExprConstant {
+                    value: Constant::Bool(value),
+                    kind: None,
+                    range: _,
+                }) = comparator
+                {
+                    match op {
+                        EqCmpop::Eq => {
```

Comment:
(But we now match here, rather than matching on `Cmpop` and then calling `op.into()` to convert.